src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -1176,12 +813,12 @@ private FunctionExpression ConvertObject(ObjectExpression expression)
         private static FunctionExpression GetCreateObjectExpression(params LanguageExpression[] parameters)
             => CreateFunction("createObject", parameters);
 
-        private LanguageExpression ConvertBinary(BinaryOperationSyntax syntax)
+        private LanguageExpression ConvertBinary(BinaryExpression binary)
         {
-            var operand1 = ConvertExpression(syntax.LeftExpression);
-            var operand2 = ConvertExpression(syntax.RightExpression);
+            var operand1 = ConvertExpression(binary.Left);
+            var operand2 = ConvertExpression(binary.Right);
 
-            return syntax.Operator switch
+            return binary.Operator switch
             {
                 BinaryOperator.LogicalOr => CreateFunction("or", operand1, operand2),
                 BinaryOperator.LogicalAnd => CreateFunction("and", operand1, operand2),
```

Comment:
Not pertinent to this PR, but we should have a static mapping of BinaryOperator => ARM function name. There's a dictionary in `Bicep.Core.TypeSystem.OperationReturnTypeEvaluator` that's very similar to this switch expression.

Reply:
Great suggestion!